fix(curriculum): require empty p with aria-live

pkdvalis opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or Gitpod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #61829

<!-- Feel free to add any additional description of changes below this line -->

required p to be present but empty
Added aria-live requirement with optional value (assertive or polite)
I also changed the HTML seed code title to OTP Generator

pkdvalis committed:
fix: require empty p with aria-live

pkdvalis committed:
fix(curriculum): empty p aria-live solution update

pkdvalis committed:
Merge branch 'main' of https://github.com/pkdvalis/freeCodeCamp into fix/otp-generator-instructions-for-p

test

pkdvalis committed:
fix: change <p> to p and ID

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md:
```diff
@@ -72,13 +72,17 @@ Initially, the `h2` inside `.container` should display the message `"Click 'Gene
 assert.exists(container, "The container element should exist");
 const pElements = document.querySelectorAll('.container p');
-assert.strictEqual(pElements.length, 0, "Initially, there should be no <p> elements inside '.container'");
+assert.strictEqual(pElements.length, 1, "There should be one <p> element inside '.container'");
```
```suggestion
assert.strictEqual(pElements.length, 1, "There should be one p element inside '.container'");
```

`p` is element, `<p>` is a tag

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md:
```diff
@@ -72,13 +72,17 @@ Initially, the `h2` inside `.container` should display the message `"Click 'Gene
 const pElements = document.querySelectorAll('.container p');
-assert.strictEqual(pElements.length, 0, "Initially, there should be no <p> elements inside '.container'");
+assert.strictEqual(pElements.length, 1, "There should be one <p> element inside '.container'");
+assert.strictEqual(pElements[0].id, "otp-timer", "The <p> element should have the id otp-timer");
```
```suggestion
assert.strictEqual(pElements[0].id, "otp-timer", "The p element should have the id otp-timer");
```

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md:
```diff
@@ -236,7 +240,7 @@ clock.uninstall()
 
 <head>
     <meta charset="UTF-8" />
-    <title>Color Picker</title>
+    <title>OTP Generator</title>
```
good catch 👍

pkdvalis committed:
Update curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md

Another thing I have noticed is that these messages do not seem to appear in the console.

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

pkdvalis committed:
Update curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

pkdvalis committed:
Update curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

ilenia-magoni commented:
I suggest, when you accept the commits, to do so from teh Files changed tab, that allows to create a batch of all the changes and a single commit

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md:
````diff
@@ -72,13 +72,17 @@ Initially, the `h2` inside `.container` should display the message `"Click 'Gene
 ```
 
-Initially, there should be a `.container` element that does not contain any `<p>` elements.
+Initially, the `.container` element should contain an empty `p` element with the ID `otp-timer` and the `aria-live` attribute set to a valid value.
````
Actually, I think this should be separated in two for better clarity

1. The `.container` erlement should contain a `p` element
2. The `p` element should be empty initially

pkdvalis replied:
How does this look?

The `.container` element should contain a `p` element with the ID `otp-timer`.

```js
const container = document.querySelector('.container');
assert.exists(container, "The container element should exist");
const pElements = document.querySelectorAll('.container p');
assert.strictEqual(pElements.length, 1, "There should be one p element inside '.container'");
assert.strictEqual(pElements[0].id, "otp-timer", "The p element should have the id otp-timer");
```

The `p` element should have an `aria-live` attribute.

```js
assert.exists(pElements[0].ariaLive, "The `p` element should have the `aria-live` attribute.");
assert.oneOf(pElements[0].ariaLive, ["assertive", "polite"], "The `aria-live` attribute should be set to a valid value.");
```

Initially, the `p` element should be empty.

```js
assert.strictEqual(pElements[0].innerText, "", "The `p` element should be empty");
```

ilenia-magoni replied:
you need to select the element in all the tests

for the second and third you can use `querySelector` as you are taking the first one and not checking if there is more

also optional chaining

---
p element exists and has id

```js
const container = document.querySelector('.container');
assert.exists(container, "The container element should exist");
const pElements = container?.querySelectorAll('p');
assert.strictEqual(pElements.length, 1, "There should be one p element inside '.container'");
assert.strictEqual(pElements[0]?.id, "otp-timer", "The p element should have the id otp-timer");
```
using `querySelectorAll` on `container` makes the search more focused

---

p element has aria-live

```js
const pElement = document.querySelector('.container p');
assert.exists(pElement?.ariaLive, "The `p` element should have the `aria-live` attribute.");
assert.oneOf(pElement?.ariaLive, ["assertive", "polite"], "The `aria-live` attribute should be set to a valid value.");
```

no need for `querySelectorAll` if you are looking for a single element

---

p element is empty

```js
const pElement = document.querySelector('.container p');
assert.strictEqual(pElement?.innerText, "", "The `p` element should be empty");
```

pkdvalis committed:
Update curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

pkdvalis committed:
fix(curriculum): separate into 3 tests

huyenltnguyen reviewed (commented):
The changes look good to me, just a few minor comments left :D

Also a quick tip, when applying suggested changes, you can apply them in bulk by clicking the "Add suggestion to batch button" and GitHub will create a single commit for the changes, rather than one commit for each change. 

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes.

pkdvalis committed:
Apply suggestions from code review

Co-authored-by: Huyen Nguyen <25715018+huyenltnguyen@users.noreply.github.com>

huyenltnguyen reviewed (approved):
LGTM. (And no more nitpicks from me 😅)

Thanks for the PR and for your patience!

pkdvalis committed:
fix(curriculum): specify arialive values in hint
